site/public/js/drag-and-drop.js

```diff
@@ -1146,10 +1146,45 @@ function handleDownloadImages(csrf_token) {
     });
 }
 
+function isValidFileName(file_name) {
```

Comment:
I think this function should be placed under `site/ts/twig/courseMaterials/upload.ts` or something, that way it can be used in other places and it would get the typescript type checking + ability to be tested/linted. 

You can then include it to the twig file via `{{ add_twig_module_js('courseMaterials/upload.js') }}`

Here is an example PR https://github.com/Submitty/Submitty/pull/7649 that uses it, the twig integration with typescript was developed by @DefinitiveAbove so he can also help out if you run into issues